### PR TITLE
Implement the Aomoto complex of the Orlik-Solomon algebra and Hilbert series for filtered modules

### PR DESCRIPTION
**Pull request by tscrim**
We implement the Aomoto complex of the Orlik-Solomon algebra $A$, which is defined as the chain complex on $A$ with the differential defined by $\omega \wedge$ for any $\omega \in A_1$.

We also provide an implemented of Hilbert series for any filtered module, where those for finite rank modules return honest polynomials. (This was originally going to be just for the Orlik-Solomon algebra, but the implementation trivially generalizes.)

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by tscrim** [comment]
@fchapoton @miguelmarco cc-ing people who might be interested.

File: src/sage/categories/filtered_modules_with_basis.py [hunk 1]
```diff
@@ -1112,3 +1158,46 @@ def maximal_degree(self):
...
+                Return the Hilbert series of ``self`` as a polynomial.
+
+                Let `R` be a commutative ring (with unit). Let
+                `M = \bigcup_{n=0}^{\infty} M_n` be a filtered `R`-module.
+                The *Hilbert series* of `M` is the formal power series
+
+                .. MATH::
+
+                    H(t) = \sum_{n=0}^{\infty} \ell(M_n) t^n,
+
```
Comment (fchapoton) [inline src/sage/categories/filtered_modules_with_basis.py:1176]: a quotient missing in the def here

File: src/sage/algebras/orlik_solomon.py [hunk 2]
```diff
@@ -537,6 +536,67 @@ def as_cdga(self):
         """
         return self.as_gca().cdg_algebra({})
 
+    def aomoto_complex(self, omega):
+        r"""
+        Return the Aomoto complex of ``self`` defined by ``omega``.
+
+        Let `A(M)` be an Orlik-Solomon algebra of a matroid `M`. Let
+        `\omega \in A(M)_1` be an element of (homoegenous) degree 1.
+        The Aomoto complete is the chain complex defined on `A(M)`
```
Comment (fchapoton) [inline src/sage/algebras/orlik_solomon.py:545]: typo homoegenous

File: src/sage/categories/filtered_modules_with_basis.py (hunk 1 above)
Comment (tscrim) [reply src/sage/categories/filtered_modules_with_basis.py:1176]: Done. I also fixed a few other details with the definition.

**Comment by tscrim** [comment]
In addition to addressing the comments, I added a doctest for the input check.

**Review by fchapoton — APPROVED** [review]
looks good. Interesting addition. Thanks

**Comment by tscrim** [comment]
Merci!

**Pull request merged**